client/src/components/ui/sliding-link-button.tsx

```diff
@@ -10,43 +10,49 @@ const SlidingLinkButton = React.forwardRef<
     buttonClassName?: string;
     Icon: React.ComponentType<React.SVGProps<SVGSVGElement>>;
     position?: 'left' | 'right' | undefined;
+    isCompact?: boolean;
   }
->(({ children, className, buttonClassName, Icon, position = 'left', ...props }, ref) => {
-  return (
-    <>
-      <Link
-        className={cn(
-          'group flex items-center justify-start',
-          { 'flex-row-reverse': position === 'right' },
-          className,
-        )}
-        ref={ref}
-        {...props}
-      >
-        <Icon
-          className={cn({
-            'mr-[15px] h-[34px] w-[34px] bg-gray-100 px-1 py-1 transition-colors group-hover:bg-slate-700 group-hover:text-white group-focus:bg-slate-700 group-focus:text-white':
-              true,
-            [buttonClassName ?? '']: !!buttonClassName,
-          })}
-        />
-        <span
+>(
+  (
+    { children, className, buttonClassName, Icon, position = 'left', isCompact = false, ...props },
+    ref,
+  ) => {
+    return (
+      <>
+        <Link
           className={cn(
-            'text-xs opacity-0 transition duration-500 group-hover:opacity-100 group-focus:opacity-100',
-            {
+            'group flex items-center justify-start',
+            { 'flex-row-reverse': position === 'right' },
+            className,
+          )}
+          ref={ref}
+          {...props}
+        >
+          <Icon
+            className={cn({
+              'mr-[15px] h-[34px] w-[34px] bg-gray-100 px-1 py-1 transition-colors group-hover:bg-slate-700 group-hover:text-white group-focus:bg-slate-700 group-focus:text-white':
+                true,
+              [buttonClassName ?? '']: !!buttonClassName,
+            })}
+          />
+          <span
+            className={cn('text-xs opacity-0 transition group-focus:opacity-100', {
+              'min-w-fit duration-500 group-hover:opacity-100': !isCompact,
+              'max-w-0 duration-0 group-hover:max-w-fit group-hover:opacity-100 group-hover:duration-500':
+                isCompact,
```

Comment:
There are some issues with the focus styles:
```suggestion
            className={cn('text-xs opacity-0 transition group-hover:opacity-100 group-focus:opacity-100', {
              'min-w-fit duration-500': !isCompact,
              'max-w-0 duration-0 group-hover:max-w-fit group-hover:duration-500 group-focus:max-w-fit group-focus:duration-500':
                isCompact,
```